### PR TITLE
Use theme transition duration for drawer

### LINKED ISSUE #210: Use recommended animation speeds for drawer
#### Describe the desired behavior
Use the recommended transition speeds for opening and closing the drawer (entering and exiting).

#### Describe the current behavior
Using the 'standard' speed for both in and out.

#### Is this request related to a current issue?
https://github.com/pxblue/react-component-library/pull/207#pullrequestreview-544130015

### PR DESCRIPTION
**Pull request by huayunh**
Fixes #210

#### Changes proposed in this Pull Request:
- Updated drawer layout and drawer to use `theme.transitions.duration.enteringScreen` and `leavingScreen`

### CODE REVIEW

File: components/src/core/DrawerLayout/DrawerLayout.tsx [hunk 1]
```diff
@@ -59,7 +65,10 @@ const DrawerLayoutRender: React.ForwardRefRenderFunction<unknown, DrawerLayoutPr
         >
             <div ref={ref} className={clsx(defaultClasses.root, classes.root)} {...otherDivProps}>
                 <div className={clsx(defaultClasses.drawer, classes.drawer)}>{drawer}</div>
-                <div className={clsx(defaultClasses.content, classes.content)} style={style}>
+                <div
+                    className={clsx(defaultClasses.content, classes.content, padding === 0 && defaultClasses.expanded)}
```
Comment (joebochill) [inline components/src/core/DrawerLayout/DrawerLayout.tsx:69]: for conditional classes in clsx, we usually prefer the object syntax:
```suggestion
                    className={clsx(defaultClasses.content, classes.content, {[defaultClasses.expanded]: padding === 0})}
```

File: components/src/core/Drawer/Drawer.tsx [hunk 2]
```diff
@@ -8,18 +8,30 @@ import { InfoListItemProps as BaseInfoListItemProps } from '../InfoListItem';
...
+        root: {
+            transition: theme.transitions.create('width', { duration: theme.transitions.duration.leavingScreen }),
+            minHeight: '100%',
+            '&$expanded': {
+                transition: theme.transitions.create('width', {
+                    duration: theme.transitions.duration.enteringScreen,
+                }),
+            },
+        },
+        expanded: {},
```
Comment (joebochill) [inline components/src/core/Drawer/Drawer.tsx:22]: New classes need to be added to the readme docs

File: components/src/core/DrawerLayout/DrawerLayout.tsx (hunk 1 above)
Comment (joebochill) [reply components/src/core/DrawerLayout/DrawerLayout.tsx:69]: applies to the one above too

**Review by joebochill — CHANGES_REQUESTED** [review]

File: components/src/core/Drawer/Drawer.tsx (hunk 2 above)
Comment (huayunh) [reply components/src/core/Drawer/Drawer.tsx:22]: Do we want to add “expanded” as part of a user class too? (Not just “defaultClasses?)

Comment (joebochill) [reply components/src/core/Drawer/Drawer.tsx:22]: I would say yes, if it's a class that gets applied, we should give them a way to customize it if desired.

**Comment by huayunh** [comment]
I just realized that DrawerLayout doesn't really have access to whether a drawer is opened or not; when the variant is set to 'persistent', padding===0 never means "drawer is collapsed". 

Perhaps we should just close this or wire the drawer's open state into the drawer layout context provider

**Comment by joebochill** [comment]
> I just realized that DrawerLayout doesn't really have access to whether a drawer is opened or not; when the variant is set to 'persistent', padding===0 never means "drawer is collapsed".
> 
> Perhaps we should just close this or wire the drawer's open state into the drawer layout context provider

Hmm good call. You're right padding=0 is not a good check for the persistent drawer. When I was messing around with the out of sync animations I implemented a pretty quick and dirty version of a context provider that shared the open state with the Layout, but it was kinda messy and I'm not sure if that's the best way to do it. We may need to have a little more discussion around it to decide on a better approach. 

You may want to add a little more context to #210 to explain the issue. 

Does this break with the persistent drawer variant (e.g., animations using different timings) or is it something that's not very noticeable?

**Comment by huayunh** [comment]
It is insignificant — I didn't even notice the classes were missing until I looked into the inspector.

**Comment by joebochill** [comment]
> It is insignificant — I didn't even notice the classes were missing until I looked into the inspector.



> It is insignificant — I didn't even notice the classes were missing until I looked into the inspector.

Then I think we can push this update in (after readme update) and make a new issue/enhancement to get the open/closed state up into the drawer layout via context (or we can hijack the existing context to pass the required padding and the open/closed state).

**Review by joebochill — APPROVED** [review]

**Pull request merged**